test/functional/tests/lazy_writes/test_lazy_writes_clean.py

```diff
@@ -0,0 +1,230 @@
+from test_tools.fs_utils import create_random_test_file
+from test_tools.iostat import IOstat
+from test_utils.filesystem.file import File
+from test_utils.os_utils import Udev
+from test_utils.size import Size, Unit
+
+bs = Size(512, Unit.KibiByte)
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_clean_stop_cache(cache_mode):
+    """
+        title: Test of the ability to stop correctly in modes with lazy writes.
+        description: |
+          Test if OpenCAS stops cache in modes with lazy writes without data loss.
+        pass_criteria:
+          - Cache stopping works properly.
+          - Data on core device is correct after cache is stopped.
+    """
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(256, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(512, Unit.MebiByte)])
+        core_part = core_dev.partitions[0]
+        Udev.disable()
+
+    with TestRun.step(f"Start cache in {cache_mode} mode."):
+        cache = casadm.start_cache(cache_part, cache_mode)
+
+    with TestRun.step("Add core to cache."):
+        core = cache.add_core(core_part)
+
+    with TestRun.step("Change cleaning policy to 'NOP'."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step("Write data to exported object."):
+        test_file_main = create_random_test_file("/tmp/test_file_main", Size(64, Unit.MebiByte))
+        dd = Dd().output(core.system_path) \
+            .input(test_file_main.full_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+
+    with TestRun.step("Check IO stats."):
+        # 4kB is written when partition is created
+        core_stats = (IOstat.get_iostat(core_part).total_writes - Size(4, Unit.KibiByte)).value
+        exp_obj_stats = IOstat.get_iostat(core).total_writes.value
+        if exp_obj_stats == 0 or core_stats > 0:
+            TestRun.LOGGER.error("Writes should occur only on cache device.")
+
+    with TestRun.step("Read data from exported object."):
+        test_file_1 = File.create_file("/tmp/test_file_1")
+        dd = Dd().output("/tmp/test_file_1") \
+            .input(core.system_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_1.refresh_item()
+
+    with TestRun.step("Compare md5 sum of test files."):
+        if test_file_main.md5sum() != test_file_1.md5sum():
+            TestRun.LOGGER.error("Md5 sums should be equal.")
+
+    with TestRun.step("Read data from core device."):
```

Comment:
I recommend to stop cache before reading data from core device, because there are cases, when system is making md5 differ when cache is running, but after stop everything is ok. What is more, test is for cache modes with lazy writes cache mode, so you should flush data before checking md5sums on core and core device.